docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +347,271 @@ recursion was explicitly requested.
+    }
+  } else {
+    assert y == 1;            // Assertion  --
+  }
+  assert y == 1;              // Assertion  Assertion
+  if !x {
+    assert y >= 1;            // Assertion  Assertion
+  } else {
+    assert y <= 1;            // Assertion  Assertion
+  }
+}
+```
+
+And another one where the focused block is guarded with a `while`, resulting in remaining assertions not being part of the first assertion batch:
+```dafny
+method doFocus2(x: bool) returns (y: int) {
+  y := 1;                     // Batch 1    Batch 2
+  assert y == 1;              // Assertion  --
+  if x {
+    while false {
+      assert y >= 0;          // --         Assertion
+      assert {:focus} y <= 2; // --         Assertion
+      y := 2;
+      assert y == 2;          // --         Assertion
+    }
+  } else {
+    assert y == 1;            // Assertion  --
+  }
+  assert y == 1;              // Assertion  --
+  if !x {
+    assert y >= 1;            // Assertion  --
+  } else {
+    assert y <= 1;            // Assertion  --
+  }
+}
+```
+
+### 22.3.2. `{:split_here}` {#sec-split_here}
+`assert {:split_here} X;` splits verification into two [assertion batches](#sec-assertion-batches).
+It verifies the code leading to this point (excluded) in a first assertion batch,
+and the code leading from this point (included) to the next `{:split_here}` or until the end in a second assertion batch.
+It might help with timeouts.
+
+Here is one example, where `--` in the comments stands for `Assumption`:
+```dafny
+method doSplitHere(x: bool) returns (y: int) {
+  y := 1;                      // Batch 1    Batch 2     Batch 3
+  assert y >= 0;               // Assertion  --          --
+  if x {
+    assert y <= 1;             // Assertion  --          --
+    assert {:split_here} true; // --         Assertion   --
+    assert y <= 2;             // --         Assertion   --
+    assert {:split_here} true; // --         --          Assertion
+    if x {
+      assert y == 1;           // --         --          Assertion
+    } else {
+      assert y >= 1;           // --         --          Assertion
+    }
+  } else {
+    assert y <= 3;             // Assertion  --          --
+  }
+  assert y >= -1;              // Assertion  --          --
+}
+```
+
+### 22.3.3. `{:subsumption n}`
+Overrides the `/subsumption` command-line setting for this assertion.
+
+## 22.4. Attributes on variable declarations
+
+### 22.4.1. `{:assumption}`
+This attribute can only be placed on a local ghost bool
+variable of a method. Its declaration cannot have a rhs, but it is
+allowed to participate as the lhs of exactly one assignment of the
+form: `b := b && expr;`. Such a variable declaration translates in the
+Boogie output to a declaration followed by an `assume b` command.
+See [@LeinoWuestholz2015], Section 3, for example uses of the `{:assumption}`
+attribute in Boogie.
 
+### 22.4.2. `{:existential}`
+Marks a global Boolean variable as existentially quantified. If
```

Comment:
There is no such thing as a global variable in Dafny, so we should leave this one out.